torch/backends/_nnapi/serializer.py

```diff
@@ -1288,6 +1298,15 @@ def add_upsample_nearest2d(self, node):
 
         out_shape = (image_oper.shape[0], image_oper.shape[1], out_h, out_w)
         use_nchw = image_oper.use_nchw()
+        out_id = self.add_tensor_operand(node.outputsAt(0), image_oper._replace(shape=out_shape))
+
+        # Handle variable input size
+        for dim in (2, 3):   # h, w indices
+            if image_oper.shape[dim] == 0:
+                if size_ctype.kind() != "NoneType":
+                    self.compute_operand_shape(out_id, dim, size_arg[dim-2])
+                elif scale_ctype.kind() != "NoneType":
+                    self.compute_operand_shape(out_id, dim, f"int({scale_arg[dim-2]} * {flex_name(image_id, dim)})")
```

Comment:
I'd suggest adding an "else" branch here that throws.  It should never be hit because of the block above, but it will make it safer against refactors.

Reply:
Fair, added